app/assets/scripts/views/task-page.js

```diff
@@ -109,7 +111,10 @@ var TaskPage = React.createClass({
       }
     }
 
-    const geometry = geometryToFeature([this.props.task.data]);
+    const geometry = {
+      'type': 'FeatureCollection',
+      'features': [geometryToFeature([this.props.task.data])]
+    };
```

Comment:
Unless I'm mistaken, the setupFeature function just needs a better conditional, and everything should be ok.
`if ((feat.features && feat.features.length) || (feat.geometry && feat.geometry.coordinates.length))`